src/tools/r2rdump/CoreDisTools.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Reflection.PortableExecutable;
+using System.Runtime.InteropServices;
+
+namespace R2RDump
+{
+    class CoreDisTools
+    {
+        public enum TargetArch
+        {
+            Target_Host, // Target is the same as host architecture
+            Target_X86,
+            Target_X64,
+            Target_Thumb,
+            Target_Arm64
+        };
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        [return: MarshalAs(UnmanagedType.I8)]
+        public static extern long InitDisasm(TargetArch Target);
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        public static extern void DumpCodeBlock(long Disasm, ulong Address, IntPtr Bytes, int Size);
+
+        [DllImport(
+        "coredistools.dll",
+        CharSet = CharSet.Ansi,
+        CallingConvention = CallingConvention.StdCall)]
+        public static extern void FinishDisasm(long Disasm);
+
+        public unsafe static void DumpCodeBlock(long Disasm, int Address, int Offset, byte[] image, int Size)
+        {
+            fixed (byte* p = image)
+            {
+                IntPtr ptr2 = (IntPtr)(p + Offset);
+                DumpCodeBlock(Disasm, (ulong)Address, ptr2, Size);
+            }
+        }
+
+        public static long GetDisasm(Machine machine)
+        {
+            TargetArch target = TargetArch.Target_Host;
+            switch (machine)
+            {
+                case Machine.IA64:
+                case Machine.Amd64:
+                    target = TargetArch.Target_X64;
+                    break;
+                case Machine.I386:
+                    target = TargetArch.Target_X86;
+                    break;
+                case Machine.Arm:
+                    target = TargetArch.Target_Arm64;
+                    break;
+                case Machine.Thumb:
```

Comment:
You can omit `case Machine.Thumb:` and just have `ArmThumb2`. `Machine.Thumb` is obsolete, it is no not used in .NET Core.